### PR TITLE
Support ltsv

### PR DESCRIPTION
**Pull request by u5surf**
### Proposed changes

Describe the use case and detail of the change. If this PR addresses an issue on GitHub, make sure to include a link to that issue using one of the [supported keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) here in this description (not in the title of the PR).

   * The many nginx's users choice the log format into Labeled Tab Separated Values to be able to analyze and  extend logs more easily.
    * So, it should be support this format that the user can be more easily to switch from conventional log collector to nginx-agent.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [x] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: src/core/tailer/tailer.go [hunk 1]
```diff
@@ -95,6 +101,28 @@ func NewPatternTailer(file string, patterns map[string]string) (*PatternTailer,
 	return &PatternTailer{t, gc}, nil
 }
 
+func NewLTSVTailer(file string) (*LTSVTailer, error) {
+	ltsvParseString := func(line string) map[string]string {
```
Comment (dhurley) [inline src/core/tailer/tailer.go:105]: Can you move this out into its own function?

File: src/core/tailer/tailer.go [hunk 2]
```diff
@@ -95,6 +101,28 @@ func NewPatternTailer(file string, patterns map[string]string) (*PatternTailer,
...
+		columns := strings.Split(line, "\t")
+		lvs := make(map[string]string)
```
Comment (dhurley) [inline src/core/tailer/tailer.go:107]: Not sure what lvs and lv mean. Could you give the variables more meaningful names?

**Review by dhurley — COMMENTED** [review]
There is a failing unit test that needs to be addressed when you get a chance. https://github.com/nginx/agent/actions/runs/5437056592/jobs/9899420232

File: src/core/tailer/tailer.go (hunk 2 above)
Comment (u5surf) [reply src/core/tailer/tailer.go]: @dhurley Also Fix those name.

File: src/core/tailer/tailer.go (hunk 1 above)
Comment (u5surf) [reply src/core/tailer/tailer.go]: @dhurley I could move out the parse function outside.

**Comment by u5surf** [comment]
> There is a failing unit test that needs to be addressed when you get a chance. https://github.com/nginx/agent/actions/runs/5437056592/jobs/9899420232

Sorry my bad, some of checksum and line count hadn’t fixed. l’ve already fixed

File: src/core/tailer/tailer_test.go [hunk 3]
```diff
@@ -137,3 +137,57 @@ T:
...
+	}
+	accessLogFile.Close()
+
+	var count int
+        var res map[string]string
+T:
+	for {
+		select {
+		case r := <-data:
+                        res = r
```
Comment (Dean-Coakley) [inline src/core/tailer/tailer_test.go:168]: Can you run `gofmt` on this file before merging?

**Review by Dean-Coakley — APPROVED** [review]

File: src/core/tailer/tailer_test.go (hunk 3 above)
Comment (u5surf) [reply src/core/tailer/tailer_test.go]: @Dean-Coakley Thanks catching.
I fixed.

**Review by oliveromahony — APPROVED** [review]

**Pull request merged**